core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -88,7 +88,9 @@ class CachedDataFrame(session: SparkSession, queryExecution: QueryExecution,
   private lazy val boundEnc = exprEnc.resolveAndBind(logicalPlan.output,
     sparkSession.sessionState.analyzer)
 
-  def queryExecutionString: String = queryExecution.toString()
+  private lazy val queryExecutionString: String = queryExecution.toString()
```

Comment:
Need to be able to generate appropriate plans for different tokens as discussed. @kneeraj can you track that issue?